### PR TITLE
[processing] Remove duplicate fill holes algorithm

### PR DESCRIPTION
**Pull request by nyalldawson**
This PR consolidates the duplicate fill holes algorithms into one algorithm, by:

- moving guts of the algorithm to QgsGeometry (with tests)
- make algorithm maintain z/m values
- make algorithm work with curved geometries
- add unit tests for algorithm

The extra Fill_holes algorithm can now safely be removed. I can't work out if processing scripts can be deprecated, but in any case when we have migration path worked out this can then be removed.

### CODE REVIEW

File: python/core/geometry/qgscurvepolygon.sip [hunk 1]
```diff
@@ -57,6 +57,14 @@ class QgsCurvePolygon: public QgsSurface
     /** Removes ring. Exterior ring is 0, first interior ring 1, ...*/
     bool removeInteriorRing( int nr );
 
+    /**
+     * Removes the interior rings from the polygon. If the minimumRingArea
+     * parameter is specified then only rings smaller than this minimum
+     * area will be removed.
+     * @note added in QGIS 3.0
+     */
+    void removeInteriorRings( double minimumRingArea = -1 );
```
Comment (nirvn) [inline python/core/geometry/qgscurvepolygon.sip]: @nyalldawson , I'm wondering whether it'd be useful to have a maximumRingArea parameter too. Say you want to simplify a feature to display nicely at a given zoomed-out level, you might want to remove rings that are below a maximumArea threshold.

Comment (nyalldawson) [inline python/core/geometry/qgscurvepolygon.sip]: That's what the minimumRingArea parameter does - it specifies the minimum allowable ring size

Comment (nyalldawson) [inline python/core/geometry/qgscurvepolygon.sip]: Fixed

**Comment by volaya** [comment]
@nyalldawson There is no option to show the script as deprecated, but i think it can safely be removed. In any case, we have API breaks like that in other places in Processing, for 3.0, so no need to worry. This is a minor issue, since the algorithm is not likely to be used in many models anyway

**Comment by volaya** [comment]
+1 from me to merge

**Comment by nyalldawson** [comment]
@volaya - ok, merged. I've also removed a duplicate create points along lines script.

**Comment by nirvn** [comment]
Nice one.

**Pull request merged**